CDCgov/prime-reportstream
# Authentication Microservice POC

This PR creates a new Spring Microservice focused on authentication via Okta and proxying requests to the right service.

Test Steps:
1. Run unit tests in auth project to ensure all is working locally
2. Ask me for staging secret for okta and set to proper environment variable locally (I'll help)
3. Ask Arnej for script to retrieve valid auth token, use it to get auth token
4. Startup this service and the submissions service
5. Call http://localhost:9000/submissions/health
6. Ensure you get a 200 response with the body "up"


## Changes
- New microservice
- Additional code added in submissions for future handling of authorization

## Checklist

### Testing

## Linked Issues
- Fixes #15623
